src/test/java/org/mifos/integrationtest/cucumber/stepdef/BulkPaymentStepDef.java

```diff
@@ -0,0 +1,168 @@
+        headers.put("X-CorrelationID", "12345678-6897-6798-6798-098765432134");
+        headers.put("Platform-TenantId", "gorilla");
+//        String fileContent = getFileContent(BPMN_FILE_URL);
+        String fileContent = getFileContent("bulk-payment.csv");
+        logger.info("file content: " + fileContent);
+        RequestSpecification requestSpec = getDefaultSpec();
+        String response =  RestAssured.given(requestSpec)
+//                .baseUri("https://bulk-connector.sandbox.fynarfin.io")
+                .baseUri("http://localhost:5002")
+                .multiPart(getMultiPart(fileContent))
+                .queryParam("type", "csv")
+                .headers(headers)
+                .expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(200).build())
+                .when()
+                .post("/batchtransactions")
+                .andReturn().asString();
+        batchId = fetchBatchId(response);
+        logger.info(batchId);
+        logger.info("Batch transaction API response: " + response);
+    }
+
+    @Given("the batch ID for the submitted CSV file")
+    public void isBatchIdAvailable(){
+        assertThat(batchId).isNotEmpty();
+    }
+
+    @When("poll the batch summary API using the batch ID")
+    public void initiateBatchSummaryApi(){
+        int retries = 5;
+        int intervalInSeconds = 30;
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Platform-TenantId", "rhino");
+        RequestSpecification requestSpec = getDefaultSpec();
+
+        for(int i = 0; i < retries; i++) {
+            String response =  RestAssured.given(requestSpec)
+                    .baseUri("http://localhost:5060")
+                    .param("batchId", batchId)
+                    .param("X-CorrelationID", "12345678-6897-6798-6798-098765432134")
+                    .headers(headers)
+                    .expect()
+                    .spec(new ResponseSpecBuilder().expectStatusCode(200).build())
+                    .when()
+                    .get("/api/v1/batch")
+                    .andReturn().asString();
+            Gson gson = new Gson();
+            BatchSummaryResponse batchSummaryResponse = gson.fromJson(response, BatchSummaryResponse.class);
+
+            logger.info("Batch summary response: " + batchSummaryResponse.toString());
+
+            logger.info("Total is {}", batchSummaryResponse.getTotal());
+            logger.info("Ongoing is {}", batchSummaryResponse.getOngoing());
+            logger.info("Successful is {}", batchSummaryResponse.getSuccessful());
+            logger.info("Failed is {}", batchSummaryResponse.getFailed());
+
+            if(batchSummaryResponse != null && batchSummaryResponse.getTotal() != 0){
+                completionPercent = (int) (batchSummaryResponse.getSuccessful()/ batchSummaryResponse.getTotal() * 100);
+            }
+
+            try {
+                Thread.sleep(intervalInSeconds * 1000);
```

Comment:
Seconds *1000 can cause a lot of delay. Did you mean milliseconds? 
Please check if there is any better option than delay. Sleep causes  test to be flaky